src/modules/package.rs

```diff
@@ -78,6 +78,19 @@ fn get_poetry_version(context: &Context, config: &PackageConfig) -> Option<Strin
     format_version(raw_version, config.version_format)
 }
 
+fn get_pep621_version(context: &Context, config: &PackageConfig) -> Option<String> {
+    let file_contents = context.read_file_from_pwd("pyproject.toml")?;
+    let pep621_toml: toml::Value = toml::from_str(&file_contents).ok()?;
+    let version = pep621_toml.get("project")?.get("version")?;
+
+    if version.is_table() {
+        None
+    } else {
+        let raw_version = version.as_str()?;
+        format_version(raw_version, config.version_format)
+    }
+}
+
```

Comment:
Can you combine this with the poetry version function to avoid reading and parsing the file twice? By the way, if both versions are present, which is expected to take precedence?

Reply:
Yeah sure that makes perfect sense. AFAIK if both a `[tool.poetry]` and a `[project]` table are present then it's an invalid `pyproject.toml` so I think in that case it's fine for starship not to show either?